### PR TITLE
docs(tooltip): add aria attributes to tooltip demo component

### LINKED ISSUE #4094: docs(tooltip): add aria-attributes to tooltip demo component
Add aria-attributes for tooltip: 
- aria-describedby to button, which opens tooltip
(The role has already been added)

source: https://www.w3.org/TR/wai-aria-practices-1.1/#tooltip

### PR DESCRIPTION
**Pull request by ghost**
Was added aria-attribute to tooltip demo component:
 - `aria-describedby` to a button, which shows a tooltip
The tooltip has an id. The value of the id and value of the `aria-describedby` (button) need to be the same.

joxi_screenshot_1522673294502

Closes #4094

### CODE REVIEW

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-describedby_attribute

MDN says that `aria-describedby` attribute must contain an ID of the element with description. In this PR you just have text like `showTooltipOnHover` and so on. This isn't an ID of the element so there's no sense in it.

W3 ARIA docs says the same:
The element that serves as the tooltip container has role tooltip.
The element that triggers the tooltip references the tooltip element with aria-describedby.

File: src/tooltip/tooltip.directive.ts [hunk 1]
```diff
@@ -12,11 +12,14 @@ import { parseTriggers } from '../utils/triggers';
...
 import 'rxjs/add/observable/timer';
 
+let newId = 0;
+
 @Directive({
   selector: '[tooltip], [tooltipHtml]',
   exportAs: 'bs-tooltip'
 })
 export class TooltipDirective implements OnInit, OnDestroy {
+  private idTooltip = newId++;
```
Comment (IlyaSurmay) [inline src/tooltip/tooltip.directive.ts]: more likely `tooltipId` or `id`

File: src/tooltip/tooltip-container.component.ts [hunk 2]
```diff
@@ -42,6 +42,8 @@ export class TooltipContainerComponent implements AfterViewInit {
   containerClass: string;
   animation: boolean;
 
+  @HostBinding('attr.id') ariaDescribedby: string;
```
Comment (IlyaSurmay) [inline src/tooltip/tooltip-container.component.ts]: attribute `id` is bound to `ariaDescribedby`? doesn't seem very logic to me. change property name to `id` or `containerId`

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

File: src/tooltip/tooltip.directive.ts [hunk 3]
```diff
@@ -248,7 +253,8 @@ export class TooltipDirective implements OnInit, OnDestroy {
         .show({
           content: this.tooltip,
           placement: this.placement,
-          containerClass: this.containerClass
+          containerClass: this.containerClass,
+          ariaDescribedby: this.ariaDescribedby
```
Comment (IlyaSurmay) [inline src/tooltip/tooltip.directive.ts]: id: this.ariaDescribedby

File: src/tooltip/tooltip-container.component.ts [hunk 4]
```diff
@@ -14,6 +14,7 @@ import { isBs3 } from '../utils/theme-provider';
     '[class]':
       '"tooltip in tooltip-" + placement + " " + "bs-tooltip-" + placement + " " + placement + " " + containerClass',
     '[class.show]': '!isBs3',
+    '[attr.id]': 'this.ariaDescribedby',
```
Comment (IlyaSurmay) [inline src/tooltip/tooltip-container.component.ts]: '[attr.id]': 'this.id',

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

File: src/tooltip/tooltip.directive.ts (hunk 3 above)
Comment (ghost) [reply src/tooltip/tooltip.directive.ts]: fixed

File: src/tooltip/tooltip.directive.ts (hunk 1 above)
Comment (ghost) [reply src/tooltip/tooltip.directive.ts]: fixed

File: src/tooltip/tooltip-container.component.ts (hunk 2 above)
Comment (ghost) [reply src/tooltip/tooltip-container.component.ts]: fixed

File: src/tooltip/tooltip-container.component.ts (hunk 4 above)
Comment (ghost) [reply src/tooltip/tooltip-container.component.ts]: fixed

File: src/tooltip/tooltip.directive.ts [hunk 5]
```diff
@@ -2,7 +2,7 @@
 import {
   Directive,
   ElementRef,
-  EventEmitter,
+  EventEmitter, HostBinding,
```
Comment (IlyaSurmay) [inline src/tooltip/tooltip.directive.ts]: please keep import syntax consistent, move `HostBinding` to a new line

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

File: src/tooltip/tooltip.directive.ts (hunk 5 above)
Comment (ghost) [reply src/tooltip/tooltip.directive.ts]: fixed

**Review by Domainv — DISMISSED** [review]

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by ludmilanesvitiy** [comment]
Tooltip demo tested in CHrome ,FF - visually nothing changes, html contain appropriate attributes and ID's. RFM

**Review by valorkin — APPROVED** [review]

**Pull request merged**